### PR TITLE
feat(FEC-9481): distributed keyboard handler to decorator

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

create keyboard decorator event to implement in each component which use keyboard event on player area.

Solves: FEC-9481

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/components/keyboard/with-keyboard-event.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+const withKeyboardEvent = (WrappedComponent: Component) => {
+  return class KeyboardComponent extends Component {
+    /**
+     * add keyboard handler
+     * @param {number} keyCode - Optional payload.
+     * @param {Function} callback - Optional payload.
+     * @returns {void}
+     *
+     * @memberof EventDispatcherComponent
```
Comment (yairans) [inline src/components/keyboard/with-keyboard-event.js]: fix

File: src/components/keyboard/keyboard-event-provider.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  _createKeyCode(key: KeyboardKey): number {
+    const altKey = key.altKey ? 1 : 0;
+    const ctrlKey = key.ctrlKey ? 1 : 0;
+    const metaKey = key.metaKey ? 1 : 0;
+    const shiftKey = key.shiftKey ? 1 : 0;
+    return parseInt('' + key.code + altKey + ctrlKey + metaKey + shiftKey);
+  }
+  /**
+   * create context player
+   * @returns {void}
```
Comment (yairans) [inline src/components/keyboard/keyboard-event-provider.js]: fix

File: src/components/keyboard/with-keyboard-event.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    addKeyboardHandler(keyCode: KeyboardKey, callback: Function): void {
+      this.context.addKeyboardHandler(keyCode, callback);
+    }
+
+    /**
+     * render component
+     *
+     * @returns {React$Element} - component element
+     * @memberof LoggerComponent
```
Comment (yairans) [inline src/components/keyboard/with-keyboard-event.js]: fix

File: src/components/language/language.js [hunk 4]
```diff
@@ -76,6 +79,38 @@ class Language extends Component {
    */
   componentDidMount() {
     this.props.eventManager.listen(document, 'click', e => this.handleClickOutside(e));
+    this.props.addKeyboardHandler({code: KeyMap.C}, event => {
+      event.preventDefault();
```
Comment (yairans) [inline src/components/language/language.js]: import this handler from `keyboard.js`. we can't maintain code like this twice.
same for all new handlers in this change

File: src/components/keyboard/keyboard-event-provider.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      this._keyboardListeners[keyCombine](event);
+    }
+  }
+  /**
+   * check if keyboard handler should be used
+   * @returns {boolean} - handler should be used
+   * @private
+   */
+  _shouldHandledKeyboardEvent(): boolean {
+    return this.props.isKeyboardEnable && !this.props.shareOverlay && !this.props.playerNav;
```
[the remark this answers is not in the document]
Comment (Yuvalke) [reply src/components/keyboard/keyboard-event-provider.js]: No, if user will change the `isKeyboardEnable` to `false` and then share will change it to `false` as well and then `true` the final state of `isKeyboardEnable` will be `true` and that's not what the user wants.

File: src/components/language/language.js (hunk 4 above)
Comment (Yuvalke) [reply src/components/language/language.js]: we'll not.
`keyboard.js` is deprecated if someone wants to use it he'll need to write the logic by himself I don't think we need to maintain this code anymore.

File: src/components/language/language.js [hunk 6]
```diff
@@ -55,9 +57,36 @@ const COMPONENT_NAME = 'Language';
 class Language extends Component {
   state: Object;
   _controlLanguageElement: any;
+  _lastActiveTextLanguage: string = '';
   // ie11 fix (FEC-7312) - don't remove
   _portal: any;
-
+  _keyboardEventHandlers: Array<KeyboardEventHandlers> = [
+    {
+      eventType: 'keydown',
```
Comment (OrenMe) [inline src/components/language/language.js]: each event should have 
{
  eventType?,
  key,
  handler
}

cause each key binding can have different eventType potentially
Also, eventType should be optional cause keyDown is the default

Comment (Yuvalke) [reply src/components/language/language.js]: I grouped by eventType to make it clear
[{eventType:keydown, 
handlers:[]}]

Comment (Yuvalke) [reply src/components/language/language.js]: Handler -> {key:keyboardKey, action:function}

**Pull request merged**